neonvm/controllers/virtualmachine_controller.go

```diff
@@ -1681,15 +1681,16 @@ func podSpec(virtualmachine *vmv1.VirtualMachine, sshSecret *corev1.Secret, conf
 // SetupWithManager sets up the controller with the Manager.
 // Note that the Runner Pod will be also watched in order to ensure its
 // desirable state on the cluster
-func (r *VirtualMachineReconciler) SetupWithManager(mgr ctrl.Manager) error {
+func (r *VirtualMachineReconciler) SetupWithManager(mgr ctrl.Manager) (func() ReconcileSnapshot, error) {
```

Comment:
I agree, had the same thought while writing it. Also not sure any better ideas — maybe split the "create the reconciler" and "setup the reconciler" steps, so that wrapping with metrics would be done by the caller? (although maybe it's iffy w.r.t. how the config is used)

Reply:
I think it would be the best approach. I thought about it when I was writing `WithMetrics` but didn't do it bc calling `WithMetrics` in the `SetupWithManager` was looking fine.

Reply:
Ok - in the interest of getting this PR in quickly, I'll do that in a follow-up PR? Can also do that here. WDYT?

Reply:
Fine with either.

Reply:
My suggestion below of ditching the callback also works :laughing:

Reply:
Ended up going with the "ditch the callback" approach :)